UML-Diagram/ExceptionErrorHandling.puml

```diff
@@ -0,0 +1,44 @@
+@startuml
+
+package "Exception Handling" {
+    enum Severity {
+        {static} + LOW
+        {static} + MEDIUM
+        {static} + HIGH
+        {static} + CRITICAL
+    }
+
+    abstract class GlobalException {
+        + Message : String
+        + Severity : Severity
+        + GlobalException(String message, Severity severity)
+    }
+
+    class BookNotFoundException {
+        + BookNotFoundException(String message)
+    }
+
+    class PatronNotFoundException {
+        + PatronNotFoundException(String message)
+    }
+
+    Exception <|-- GlobalException
+    GlobalException <|-- BookNotFoundException
+    GlobalException <|-- PatronNotFoundException
+    GlobalException --> Severity
+}
+
+package "Error Handling" {
+    class ILogService {
+        + LogError(Severity severity, string message) : void
+    }
+
+    class LogService implements ILogService {
+        + LogError(Severity severity, string message) : void
+        + GenerateLogFile() : void
+    }
+
+    ILogService --> Severity
```

Comment:
i think its missed, were this implementation its going to be used, i think maybe in console system in some menus, in some actions.

Reply:
I'd like to put in the some menus as you mention but I prefer to implement it while it's working in the code due to see where it goes better.